src/iop/rawprepare.c

```diff
@@ -574,8 +574,10 @@ int process_cl(
     const float rel_to_map[2] = { 1.0f / d->gainmaps[0]->map_spacing_h, 1.0f / d->gainmaps[0]->map_spacing_v };
     const float map_origin[2] = { d->gainmaps[0]->map_origin_h, d->gainmaps[0]->map_origin_v };
 
+    for(int i = 0; i < 4; i++) dev_gainmap[i] = NULL;
```

Comment:
Why is this needed?
I suppose that dt_opencl_alloc_device should be returning NULL when it fails to allocate.

Reply:
Yes - i have thought so too for a long time and the code all over the dt code implies this. Unfortunately this is not true. The CL specs define something like allowed lazy allocation, you will get a valid pointer but the actual allocation will be done at runtime while using it in a kernel. This is vendor depending ...

Needed? you are right as we define them to be NULL above :-)